### PR TITLE
Added Chat sample app that uses new Azure Core

### LINKED ISSUE #643: Update the sample to evaluate convinence-layer
Once the task https://github.com/Azure/azure-sdk-for-android/issues/641 is complete, update the sample introduced in https://github.com/Azure/azure-sdk-for-android/issues/640 to use convinence-layer.

### PR DESCRIPTION
**Pull request by vcolin7**
Fixes #643

Support for listening for real-time notifications via Trouter is pending. Pagination methods are pending as well, since they have not been added to the Chat convenience layer in this branch.

### CODE REVIEW

File: samples/sample-chat-app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,44 @@
+plugins {
+    id 'com.android.application'
+}
+
+android {
+    compileSdkVersion 30
+    buildToolsVersion "30.0.2"
```
Comment (anuchandy) [inline samples/sample-chat-app/build.gradle:7]: use buildToolsVersin `30.0.3`

**Review by anuchandy — COMMENTED** [review]
To resolve the build failure, you can try pull from upstream (v2-sdk-acs), I've added necessary entries to parent gradle to take care of jcenter shutdown;

File: samples/sample-chat-app/src/main/java/com/azure/android/communication/chat/sampleapp/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+
+    private void logAndToast(String msg) {
+        Log.d(TAG, msg);
+        Toast.makeText(this, msg, Toast.LENGTH_LONG).show();
+    }
+
+    public void startRealTimeNotification(View view) {
+        logAndToast("Implementation pending");
+
+        /*Log.i(TAG, "Starting real time notification");
```
Comment (marche0133) [inline samples/sample-chat-app/src/main/java/com/azure/android/communication/chat/sampleapp/MainActivity.java:131]: Mark TODO if we would like to keep these commented code

File: samples/sample-chat-app/src/main/java/com/azure/android/communication/chat/sampleapp/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+    //private JSONObject eventPayload;
+
+    // Replace firstUserId and secondUserId with valid communication user identifiers from your ACS instance.
+    private String firstUserId = "<first-user-id>";
+    private String secondUserId = "<second-useirer-id>";
+    // Replace userAccessToken with a valid communication service token for your ACS instance.
+    private final String firstUserAccessToken = "<first-user-access-token>";
+    private String threadId = "<to-be-updated-below>";
+    private String chatMessageId = "<to-be-updated-below>";
+    private final String endpoint = "https://vicolina-acs.communication.azure.com";
```
[the remark this answers is not in the document]
Comment (vcolin7) [reply samples/sample-chat-app/src/main/java/com/azure/android/communication/chat/sampleapp/MainActivity.java:51]: Thank you, I thought I had removed it but apparently not :P

**Review by marche0133 — APPROVED** [review]

**Pull request merged**